open-source-hosts-plugin/scripts/AwsTools/SumerianHostScript.ts

```diff
@@ -0,0 +1,207 @@
+});
+ ```
+ */
+
+type SumerianHostMetadata = {
+  bindPoseOffsetName: string;
+  poiConfigPath: string;
+  gestureConfigPath: string;
+  animClipPaths: any;
+  lookJoint: string;
+};
+
+// Our Github Actions will replace this with a commit SHA at release time
+// Right now this script is the only runtime asset published by our plugin
+// As we build out more runtime complexity into this plugin, this versioning should move there
+export const PLUGIN_VERSION = 'development';
+
+export default class SumerianHostScript extends Mesh {
+  @visibleInInspector('string', 'Voice ID', 'Joanna')
+  public pollyVoiceId = 'Joanna';
+
+  @visibleInInspector('string', 'Language ID', 'en-US')
+  public pollyLanguageId = 'en-US';
+
+  @visibleInInspector(
+    'string',
+    'Polly Engine ("standard" or "neural")',
+    'neural'
+  )
+  public pollyEngine = 'neural';
+
+  @fromScene('AWSConnector')
+  public awsConnector: IAwsConnector;
+
+  /**
+   * Observer which signals when the SumerianHost.host instance is fully
+   * initialized and ready for use.
+   */
+  public onHostReadyObserver: Observable<any> = new Observable();
+
+  /**
+   * Provides access to the APIs that control host functionality, namely:
+   *
+   * Text-to-speech functions, available via `SumerianHost.host.TextToSpeechFeature`.
+   * See [TextToSpeechFeature API documentation](https://aws-samples.github.io/amazon-sumerian-hosts/babylonjs_TextToSpeechFeature.html).
+   *
+   * Point-of-interest tracking, available via `SumerianHost.host.PointOfInterestFeature`.
+   * See [PointOfInterestFeature API documentation](https://aws-samples.github.io/amazon-sumerian-hosts/babylonjs_PointOfInterestFeature.html).
+   *
+   * Gesture functions, available via `SumerianHost.host.GestureFeature`.
+   * See [GestureFeature API documentation](https://aws-samples.github.io/amazon-sumerian-hosts/core_GestureFeature.html)
+   *
+   * Developer note: This property is typed as "HostObject | any" because
+   * the HostObject's TypeScript definition doesn't include some properties
+   * that are dynamically added to the object at runtime such as
+   * HostObject.TextToSpeechFeature and others. The "any" designation
+   * prevents the TypeScript compiler from reporting errors when accessing those
+   * dynamic properties.
+   */
+  public host: HostObject | any;
+
+  // @ts-ignore DO NOT EDIT this empty constructor!
+  protected constructor() {}
+
+  /**
+   * This method gets called immediately after the constructor.
+   */
+  public async onInitialize(): Promise<void> {
+    const config: SumerianHostMetadata = this.getMetadata();
+    const poiConfig = await HostObject.loadJson(config.poiConfigPath);
+
+    await this.initHostCharacter(config, poiConfig);
+    this.initPointOfInterestTracking(poiConfig, config.lookJoint);
+    await this.initTextToSpeech();
+    this.onHostReady();
+  }
+
+  protected async initHostCharacter(
+    config: any,
+    poiConfig: any
```

Comment:
Do we have types we can use for these?

Reply:
I'll create some.

Reply:
This is now fixed in the latest PR update.